### ISSUE TITLE
vmalert: failed to send alerts to addr ... invalid SC 201

### ISSUE THREAD

**Issue by vcill** [VictoriaMetrics/VictoriaMetrics#6110]
### Describe the bug

- I'm using vmalert to send alerts to Alerta (https://github.com/alerta/alerta) via webhook
- For each sucessfull POST request, Alerta returns status code `201 Created`
- Apparently (see the attached logs), this is not considered a success by vmalert 

I think any response code in the range 200-299 should be considered a success (as per HTTP protocol specs). What do you guys think? 

### Version

victoria-metrics-20240404-005236-tags-v1.100.0-0-ge9da0b1714

### Logs

error   VictoriaMetrics/app/vmalert/rule/group.go:363   group "ReverseProxy alerts": errors(1): rule "ReverseProxyNotRunning alert": failed to send alerts to addr "http://localhost:8000/webhooks/alerta-vmalert/api/v2/alerts": invalid SC 201 from "http://localhost:8000/webhooks/alerta-vmalert/api/v2/alerts"; response body: {"alert":{"attributes":{"endsAt":"2024-04-15T18:54:08.836384483+02:00","externalUrl":"http://10.220.24.112:9093","moreInfo":"<a href=\"http://lc-ems-1-2:8880/vmalert/alert?group_id=5428157339941378133&alert_id=17462675242479737900\" target=\"_blank\">Prometheu..running\"}}","receiveTime":"2024-03-15T09:41:39.134Z","repeat":true,"resource":"g1ber-lb-2","service":["reverse-proxy"],"severity":"major","status":"open","tags":["alertname=ReverseProxyNotRunning alert","c5dc=berlin","job=reverseproxy-berlin","alertgroup=ReverseProxy alerts"],"text":"Apache process not running","timeout":86400,"trendIndication":"moreSevere","type":"prometheusAlert","updateTime":"2024-03-15T09:41:39.134Z","value":null},"id":"a73a2672-94e0-43c2-9a4e-c18b312b6b69","status":"ok"}

### Used command-line flags

```
/usr/local/bin/vmalert \
...
-notifier.url=http://localhost:8000/webhooks/alerta-vmalert \
-notifier.basicAuth.username=admin \
-notifier.basicAuth.password=secret
...
```

**Comment by Amper** [comment]
Hey @vcill, this behavior will be fixed in [this PR](https://github.com/VictoriaMetrics/VictoriaMetrics/pull/6111).

**Comment by vcill** [comment]
Cool, thank you guys for the great work!

**Comment by valyala** [comment]
[`vmalert`](https://docs.victoriametrics.com/vmalert/) should properly handle `-notifier.url` response status codes in the range `200-299` after the commit 7308bad777599508db614c6b5ffd782c490ce002 . This commit will be included in the next release. In the mean time it is possible to build `vmalert` from this commit according to [these docs](https://docs.victoriametrics.com/vmalert/#how-to-build-from-sources) and verify whether it properly handles response status codes from [Alerta](https://github.com/alerta/alerta).

**Comment by valyala** [comment]
FYI, the bugfix for this issue has been included in [v1.93.14 LTS release](https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v1.93.14).

Re-opening this issue until the bugfix is included in the upcoming latest release.

**Comment by valyala** [comment]
The bugfix has been also included in [v1.97.4 LTS release](https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v1.97.4).

**Comment by hagen1778** [comment]
This bugfix was included in [v1.101.0](https://docs.victoriametrics.com/changelog/#v11010) release.
